catkin_ws/src/planner/src/substates/utility/controller.py

```diff
@@ -109,6 +114,26 @@ def get_state_goal(self,state,keepers,displace):
         goal.displace = displace
         goal.do_x, goal.do_y, goal.do_z, goal.do_theta_x, goal.do_theta_y, goal.do_theta_z = keepers
         return goal
+    
+    def pose_goal(self,position,quaternion,keepers,displace):
+        x,y,z = position
+        w,qx,qy,qz = quaternion
+        goal = QuaternionGoal()
+        goal.pose.position.x = x
+        goal.pose.position.y = y
+        goal.pose.position.z = z
+        goal.pose.orientation.w = w
+        goal.pose.orientation.x = qx
+        goal.pose.orientation.y = qy
+        goal.pose.orientation.z = qz
+        goal.displace = displace
+        goal.do_x, goal.do_y, goal.do_z, goal.do_quaternion = keepers
```

Comment:
still unclear whether or not quaternion server does position as well as quaternion rotation. Wouldn't it be simpler and easier to read if there were 3 servers: quaternion, position, and effort?

Reply:
For now we would still need a server that accommodates euler rotations until we can fully trust the quaternion code

Reply:
if Quaternion does x,y,z and rotation then we really need to A. integrate the goal sending to Quaternion server into the controller interface functions and B. remove any trace of the old server in planner (we can keep the class definition as a backup or make a backup of the current state of noetic to a branch but having 3 servers in the controller is confusing af)

If not then it shouldn't take in a Pose and we should make that clearer

Reply:
> 

I would hold off on merging then until we are sure it works at least to a certain extent. Gonna be so messy otherwise

Reply:
We could do that but it would probably introduce like a million bugs